src/model/fileinfo.py

```diff
@@ -1,11 +1,88 @@
-# TODO: Create FileInfo class. It should contain the following private variables:
-#  - __prog_path:           str
-#  - __prog_name:           str
-#  - __prog_description:    str
-#  - __args:                List[Argument]
-#  - __is_main_runnable:    bool
-
-# TODO: Create constructor for FileInfo class, that sets all variables above.
-# TODO: Create getters for all the variables in FileInfo class.
-# TODO: Create add_argument(arg: Argument) method.
-# TODO: Create setter function for __is_main_runnable variable.
+from typing import List # for the argument list
+from argument import Argument  # to have access to argument class
+
+class FileInfo:    
+    """
+    Contains all the data about a runnable.
+    
+    Attributes:
+    - __prog_path:           the path to the runnable
+    - __prog_name:           the name of the runnable
+    - __prog_description:    the description of a runnable
+    - __args:                the argument list of the runnable
+    - __is_main_runnable:    indicates if it is a main runnable
+    """
+    def __init__(self, prog_path: str, prog_name: str, prog_description: str, args: List[Argument], is_main_runnable: bool):
+        self.__prog_path = prog_path
+        self.__prog_name = prog_name
+        self.__prog_description = prog_description
+        self.__args = args
+        self.__is_main_runnable = is_main_runnable
+
+    def get_prog_path(self):
```

Comment:
This looks really great, just please make one addition.
Start the getters with @property
For example:
@property
def get_prog_path...

This addition makes readonly the return value. Otherwise you return a refernece.

Reply:
I already made a fix for that as I recongized i missed those. Or does it matter if it's @property or @Property?